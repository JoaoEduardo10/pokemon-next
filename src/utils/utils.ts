//filtragem apiPokemon

export type Tipo = {
    slot: number
    type: {
        name: string
        url: string
    }
}

export type PropsPokemonFiltro = {
    name: string
    id: number
    height: number
    weight: number
    types: Tipo[]
}

export const filtroPokemon = (pokemon: any) => {
	const { name, id, types, height, weight }: PropsPokemonFiltro = pokemon;

	return {
		name,
		id,
		types,
		height,
		weight
	};
};