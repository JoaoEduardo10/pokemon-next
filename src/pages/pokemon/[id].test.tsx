import PokemonId  from './[id]';
import { render, screen } from '@testing-library/react';
import { PropsPokemonFiltro } from '../../utils/utils';

const mockPokemon: PropsPokemonFiltro = {
	height: 100,
	id: 1,
	name: 'picachu',
	types: [
		{
			slot: 1,
			type: {
				name: 'Fire',
				url:'https://pokemon'
			}
		}
	],
	weight: 100,
};

describe('<PokemonId />', () => {
	it('should render a pokemon', () => {
		const { container } = render(<PokemonId pokemon={mockPokemon} />);

		expect(screen.getByRole('img', { name: mockPokemon.name })).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});