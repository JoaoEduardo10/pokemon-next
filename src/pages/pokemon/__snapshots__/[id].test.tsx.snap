// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonId /> should render a pokemon 1`] = `
<div>
  <section
    class="pokemon_conteiner"
  >
    <h1
      class="pokemon_title"
    >
      picachu
    </h1>
    <img
      alt="picachu"
      data-nimg="1"
      decoding="async"
      height="180"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75 2x"
      style="color: transparent;"
      width="180"
    />
    <div>
      <h3>
        Numero:
      </h3>
      <p>
        #
        1
      </p>
    </div>
    <div>
      <h3>
        Tipos:
      </h3>
      <ul
        class="types_conteiner"
      >
        <li
          class="type type_Fire"
        >
          Fire
        </li>
      </ul>
    </div>
    <div
      class="data_conteiner"
    >
      <div
        class="data_height"
      >
        <h4>
          Altura:
        </h4>
        <p>
          1000.0
           cm
        </p>
      </div>
      <div
        class="data_weight"
      >
        <h4>
          Peso:
        </h4>
        <p>
          10.0
           kg
        </p>
      </div>
    </div>
  </section>
</div>
`;
